scripts/Install-VSCode.ps1

```diff
@@ -133,118 +133,400 @@ param(
     [ValidateSet("Stable", "Insider-System", "Insider-User")]
     [string]$BuildEdition = "Stable",
 
-    [Parameter()]
-    [switch]$User,
-
     [Parameter()]
     [ValidateNotNull()]
     [string[]]$AdditionalExtensions = @(),
 
     [switch]$LaunchWhenDone,
 
-    [switch]$EnableContextMenus
+    [switch]$EnableContextMenus,
+
+    [switch]$WhatIf
 )
 
-if (($PSVersionTable.PSVersion.Major -le 5) -or $IsWindows) {
-    switch ($Architecture) {
-        "64-bit" {
-            if ((Get-CimInstance -ClassName Win32_OperatingSystem).OSArchitecture -eq "64-bit") {
-                $codePath = $env:ProgramFiles
-                $bitVersion = "win32-x64"
+function Test-IsOsX64 {
+    if ($PSVersionTable.PSVersion.Major -lt 6) {
+        return (Get-CimInstance -ClassName Win32_OperatingSystem).OSArchitecture -eq "64-bit"
+    }
+
+    return [System.Runtime.InteropServices.RuntimeInformation]::OSArchitecture -eq [System.Runtime.InteropServices.Architecture]::X64
+}
+
+function Get-LinuxReleaseInfo {
+    if (-not (Test-Path '/etc/*-release')) {
+        return $null
+    }
+
+    return Get-Content -Raw '/etc/*-release' -ErrorAction SilentlyContinue `
+        | ConvertFrom-Csv -Delimiter '=' -Header 'Key','Value' `
+        | ForEach-Object { $obj = @{} } { $obj[$_.Key] = $_.Value } { [pscustomobject]$obj }
+}
+
+function Get-CodePlatformInformation {
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('32-bit', '64-bit')]
+        [string]
+        $Bitness,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Stable', 'Insider-System', 'Insider-User')]
+        [string]
+        $BuildEdition
+    )
+
+    if ($IsWindows -or $PSVersionTable.PSVersion.Major -lt 6) {
+        $os = 'Windows'
+    }
+    elseif ($IsLinux) {
+        $os = 'Linux'
+    }
+    elseif ($IsMacOS) {
+        $os = 'MacOS'
+    }
+    else {
+        throw 'Could not identify operating system'
+    }
+
+    if ($Bitness -ne '64-bit' -and $os -ne 'Windows') {
+        throw "Non-64-bit *nix systems are not supported"
+    }
+
+    if ($BuildEdition.EndsWith('User') -and $os -ne 'Windows') {
+        throw 'User builds are not available for non-Windows systems'
+    }
+
+    switch ($BuildEdition) {
+        'Stable' {
+            $appName = "Visual Studio Code ($Bitness)"
+            break
+        }
+
+        'Insider-System' {
+            $appName = "Visual Studio Code - Insiders Edition ($Bitness)"
+            break
+        }
+
+        'Insider-User' {
+            $appName = "Visual Studio Code - Insiders Edition ($($Architecture) - User)"
+            break
+        }
+    }
+
+    switch ($os) {
+        'Linux' {
+            $releaseInfo = Get-LinuxReleaseInfo
+
+            switch ($releaseInfo.NAME) {
+                { 'Ubuntu','Debian' -contains $_ } {
```

Comment:
Rip elementary lol If we can confirm that it works on elementary, I'll send a PR... I guess you cover general Linux below

Reply:
I actually thought about just testing if `apt`/`dpkg` vs `rpm` is available... Maybe we should do that instead?